lib/net/http/response.rb

```diff
@@ -40,7 +40,7 @@ def read_new(sock)   #:nodoc: internal use only
 
     def read_status_line(sock)
       str = sock.readline
-      m = /\AHTTP(?:\/(\d+\.\d+))?\s+(\d\d\d)(?:\s+(.*))?\z/in.match(str) or
+      m = /\AHTTP(?:\/(\d+\.?\d*))?\s+(\d\d\d)(?:\s+(.*))?\z/in.match(str) or
```

Comment:
This regexp has an exponential backtracking bomb, and can match "DIGIT." without digits after a DOT.

Reply:
Hey @nobu !

thanks for your input.. you are right.. I made the change.. Can you verify if it is ok?

Thanks so much!

Reply:
no.. the change is wrong.. I will fix after and ping you again!

thanks

Reply:
hey @nobu thanks for your input!

I updated the PR.. can you re-review?

thanks

Reply:
You could use an un-captured grouping `(?:\.\d+)?` not to affect `m.captures`.